napari/layers/labels/labels.py

```diff
@@ -465,9 +506,20 @@ def _raw_to_displayed(self, raw):
         image : array
             Image mapped between 0 and 1 to be displayed.
         """
-        image = np.where(
-            raw > 0, colormaps._low_discrepancy_image(raw, self._seed), 0
-        )
+        if self.enable_custom_color_dict:
+            u, inv = np.unique(raw, return_inverse=True)
+            image = np.array(
+                [
+                    self.custom_color[x]
+                    if x in self.custom_color
+                    else colormaps._low_discrepancy_image(x, self._seed)
+                    for x in u
+                ]
+            )[inv].reshape(raw.shape)
```

Comment:
This works quite nicely. I just did some timing checks using
```python
 image = np.vectorize(lambda x: self.custom_color.get(x, 1.0))(raw)
```
and 
```python
 image = np.vectorize(self.custom_color.__getitem__)(raw)
```
both of which would have been my first impulse (note for the second one you need to make sure that all the labels are in that dict), but in both cases they are slower than the np.unique/ inv approach. I found this helpful comment give some hints to why https://stackoverflow.com/a/16993364 but it makes sense, as we are in a regime here where I think the number of unique variables will be very low, so very nice implementation!

Reply:
(I didn't expect you caught this, this certainly is beyond my python knowledge, I found it in a stack overflow post :) )